File: delcordra.py
import getpass
import cordra
from alive_progress import alive_bar

local = input("Local?(y/n) ")

if local == 'y':
    host = 'https://localhost:8443'
    username = 'admin'
    password = getpass.getpass()
    token = cordra.Token.create(host, username, password, verify = False)
else:
    host = input("Cordra's host: ")
    username = input("Username: ")
    password = getpass.getpass()
    token = cordra.Token.create(host, username, password, verify = False)


r = cordra.CordraObject.find(host, '/CamiloExplore:1', ids = True, token = token, verify = False, full = True)
ids = r["results"]
with alive_bar(len(ids), title = "Deleting objects", spinner = "arrow", bar ="classic") as bar:
    for x in ids:
        cordra.CordraObject.delete(host, x, token = token, verify = False)
        bar()

File: delcordra.jl
using CordraClient
using ProgressBars
using JSON

print("Local? (y/n) ")
l = readline()

if l == "y"
    host = "https://localhost:8443"
    username = "admin"
else
    host = "https://sandbox.materialhub.org"
    username="camilovelezr"
end

cc = CordraConnection(host, username, verify = false)

r = find_object(cc, "/CamiloExplore:1"; ids = true, pageSize = -1)

@time begin
    Threads.@threads for id in ProgressBar(r["results"])
        delete_object(cc, id)
    end
end

#@JSON read_object(cc, "test/gsr-2019-40c6/mllsq_generic_gsr")

